### PR TITLE
Reduce the logs

### PR DESCRIPTION
**Pull request by olga24912**
* fix:https://github.com/Near-One/btc-light-client-contract/issues/46

### CODE REVIEW

**Comment by kiseln** [comment]
What is the log limit?

**Comment by karim-en** [comment]
> What is the log limit?

100

**Review by kiseln — APPROVED** [review]

File: contract/src/lib.rs [hunk 1]
```diff
@@ -307,7 +305,7 @@ impl BtcLightClient {
 
         let current_header = ExtendedHeader {
             block_header,
-            block_hash: current_block_hash,
+            block_hash: current_block_hash.clone(),
```
Comment (karim-en) [inline contract/src/lib.rs:308]:
```suggestion
            block_hash: current_block_hash,
```

extra clone

File: contract/src/lib.rs [hunk 2]
```diff
@@ -316,7 +314,7 @@ impl BtcLightClient {
         if prev_block_header.block_hash == self.mainchain_tip_blockhash {
             // Probably we should check if it is not in a mainchain?
             // chainwork > highScore
-            log!("Saving to mainchain");
+            log!("Block {}: saving to mainchain", current_block_hash);
```
Comment (karim-en) [inline contract/src/lib.rs:317]:
```suggestion
            log!("Block {}: saving to mainchain", current_header.block_hash);
```

File: contract/src/lib.rs [hunk 3]
```diff
@@ -326,6 +324,7 @@ impl BtcLightClient {
             self.mainchain_tip_blockhash = current_header.block_hash.clone();
             self.store_block_header(current_header);
         } else {
+            log!("Block {}: saving to fork", current_block_hash);
```
Comment (karim-en) [inline contract/src/lib.rs:327]:
```suggestion
            log!("Block {}: saving to fork", current_header.block_hash);
```

**Review by karim-en — CHANGES_REQUESTED** [review]

File: contract/src/lib.rs (hunk 1 above)
Comment (olga24912) [reply contract/src/lib.rs]: fixed: https://github.com/Near-One/btc-light-client-contract/pull/53/commits/da81ff4eab50e113bcd7960b3f11238c539430fe

File: contract/src/lib.rs (hunk 2 above)
Comment (olga24912) [reply contract/src/lib.rs]: fixed: https://github.com/Near-One/btc-light-client-contract/pull/53/commits/da81ff4eab50e113bcd7960b3f11238c539430fe

File: contract/src/lib.rs (hunk 3 above)
Comment (olga24912) [reply contract/src/lib.rs]: fixed: https://github.com/Near-One/btc-light-client-contract/pull/53/commits/da81ff4eab50e113bcd7960b3f11238c539430fe

**Pull request merged**